### PR TITLE
[27.x] Even more backports

### PR DESCRIPTION
**Pull request by fanquake**
Backports:
* https://github.com/bitcoin/bitcoin/pull/30552
* https://github.com/bitcoin/bitcoin/pull/30633

### CODE REVIEW

File: doc/release-notes.md [hunk 1]
```diff
@@ -74,6 +82,7 @@ Thanks to everyone who directly contributed to this release:
 - Cory Fields
 - Martin Zumsande
 - Max Edwards
+- Peter Todd
```
Comment (petertodd) [inline doc/release-notes.md:85]: Worth crediting @1440000bytes too here, as they finalized my pull-req.

**Review by Theschorpioen — APPROVED** [review]

**Comment by ariard** [comment]
I would recommend to re-add in the backport release notes that excerpt from `24.x`:
https://github.com/bitcoin/bitcoin/blob/master/doc/release-notes/release-notes-24.0.1.md

```
Contributors to this project strongly recommend that merchants and services
not accept unconfirmed transactions as final, and if they insist on doing so,
to take the appropriate steps to ensure they have some recourse or plan for
when their assumptions do not hold.
```

**Comment by petertodd** [comment]
@ariard Honestly I'm inclined to think that warning isn't necessary anymore. 24.x was two years ago, and anyone crazy enough to rely on first-seen has had plenty of warning; I'm not actually aware of any merchants or services still accepting unconfirmed transactions.

The last one I was aware of was the Chivo Bitcoin ATMs in El Salvador... but they would accept unconfirmed transactions even with BIP-125 opt-in enabled, so that example isn't actually relevant to this change.

If anything, I think that wording would actually confuse the issue as you could read it as though there was still a controversy about how safe unconfirmed transactions were. If you were to put in a warning, it should say something like "Bitcoin Core reminds people that unconfirmed transactions can always be cancelled by the sender. Only after at least one confirmation can a transaction be relied upon."

**Comment by ariard** [comment]
@petertodd, Yeah it might be too much zeal, yet I’ve been surprised when the option was introduced with `24.x` how much services were still claiming to rely zero-conf acceptance. Same, I’m not aware of any bitcoin industry traffic with real-traffic actually accepting unconfirmed transactions (apart of few LSPs with zero-conf, it’s a paradigm of its own and people running that are more savvy about unconf txn risks due to channel funding flow).

The proposed alternative wording “Bitcoin Core reminds people that unconfirmed transactions can always be cancelled by the sender. Only after at least one confirmation can a transaction be relied upon” sounds indeed better.

File: doc/release-notes.md (hunk 1 above)
Comment (fanquake) [reply doc/release-notes.md:87]: Added.

**Review by stickies-v — COMMENTED** [review]
code review ACK https://github.com/bitcoin/bitcoin/commit/885c4b7bc65c294b4b2fac91392d23a37a1f6485 but 8932090f9ecd933c86ee7800c0858e8d1c0bd40b is missing backport metadata in the commit message.

I verified that all backport commits are clean, except:

https://github.com/bitcoin/bitcoin/commit/8932090f9ecd933c86ee7800c0858e8d1c0bd40b backported from https://github.com/bitcoin/bitcoin/commit/590456e3f1043ba0680e0afec9fd7653db1098bb: merge conflicts from 539404fe0f and 8950053636 which have been handled in a minimal and straightforward way.

I think backporting full-rbf makes sense conceptually, yet at the same time I'm a bit worried that shipping a new default value might be considered controversial for maintenance releases where I think people expect to be able to upgrade "automatically" without (non-bug) behaviour change? Although, of course, this exception already exists for consensus changes. Just putting it out there, no strong view.

**Comment by petertodd** [comment]
The cases where enabling full-rbf causes problems are very rare: you need to be running a service that is _not_ vulnerable to unconfirmed double-spends in general. Yet _does_ have some kind of bug related to double spends. And those rare services can still easily turn full-rbf off with a single config setting.

Meanwhile, the harm to compact block propagation caused by not enabling full-rbf can be seen as a bug that should be fixed.

**Comment by petertodd** [comment]
> Yeah it might be too much zeal, yet I’ve been surprised when the option was introduced with `24.x` how much services were still claiming to rely zero-conf acceptance.

Only one service, GAP600, claimed to still rely on zero-conf and they refused to actually provide any examples of clients actually using them. I think it's highly likely that GAP600 was actually a scam that took peoples' money and provided nothing of value. Us continuing to claim that zeroconf services still exist is actually helping perpetrate scams like GAP600.

> Same, I’m not aware of any bitcoin industry traffic with real-traffic actually accepting unconfirmed transactions (apart of few LSPs with zero-conf, it’s a paradigm of its own and people running that are more savvy about unconf txn risks due to channel funding flow).

Zeroconf LSPs are a very different thing. The trust model is explicitly trusting the LSP not to double spend you while the channel is unconfirmed. Phoenix, as an example, uses the BIP125 opt-in bit on their zeroconf channel transactions, so full-RBF settings are irrelevant.

**Comment by fanquake** [comment]
I'm just going to backport the other two fixes here, and will follow up with any other changes before the next 27.x.

**Comment by petertodd** [comment]
@fanquake That's perfectly reasonable. Concept ACK.

**Review by stickies-v — APPROVED** [review]
ACK b06c4c6550545351610fc3278dffdd63d5954cf8

File: doc/release-notes.md (hunk 1 above)
Comment (1440000bytes) [reply doc/release-notes.md]: Its weird that it was added in release notes for 27.2 but not 28.0.

Anyway, it doesn't really make any difference. Just shows the lack of courtesy.

Cc: @jonatack

**Pull request merged**